src/main/java/io/jenkins/plugins/sample/OnboardingTaskBuilder.java

```diff
@@ -0,0 +1,257 @@
+import jenkins.model.GlobalConfiguration;
+import jenkins.tasks.SimpleBuildStep;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.servlet.ServletException;
+import java.io.*;
+import java.lang.reflect.Type;
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class OnboardingTaskBuilder extends Builder implements SimpleBuildStep {
+
+    private static final String CATEGORY_JOBS_FILE = "category_jobs.json";
+    private static final Gson gson = new GsonBuilder().registerTypeAdapter(Instant.class, new InstantAdapter())
+            .setPrettyPrinting().create();
+    private static final Object lock = new Object();
+
+    public static class InstantAdapter extends TypeAdapter<Instant> {
+        @Override
+        public void write(JsonWriter out, Instant value) throws IOException {
+            out.value(value.toString());
+        }
+
+        @Override
+        public Instant read(JsonReader in) throws IOException {
+            return Instant.parse(in.nextString());
+        }
+    }
+
+    private final String name;
+    private String selectedCategory;
+    @DataBoundConstructor
+    public OnboardingTaskBuilder(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    @DataBoundSetter
+    public void setSelectedCategory(String category) {
+         this.selectedCategory = category;
+    }
+
+    public String getSelectedCategory() {
+        return selectedCategory;
+    }
+
+    @Override
+    public void perform(Run<?, ?> run, FilePath workspace, EnvVars env, Launcher launcher, TaskListener listener)
+            throws InterruptedException, IOException {
+        OnboardingSectionConfiguration onboardingSectionConfig = GlobalConfiguration.all()
+                .get(OnboardingSectionConfiguration.class);
+        if (Objects.nonNull(onboardingSectionConfig)) {
+            Optional<CategoryEntry> categoryEntryOptional = onboardingSectionConfig.getCategoryConfig().getCategories()
+                    .stream()
+                    .filter(category -> category.getUuid().equals(selectedCategory)).findFirst();
+            listener.getLogger().println("I'm " + name + (categoryEntryOptional
+                    .map(categoryEntry -> " and I have selected " + categoryEntry.getName()).orElse("")));
+            categoryEntryOptional.ifPresent(categoryEntry -> saveJobNamePerCategory(run.getFullDisplayName(), categoryEntry.getName()));
+        }
+    }
+
+    private void saveJobNamePerCategory(String jobName, String category) {
+        File file = new File(CATEGORY_JOBS_FILE);
+        synchronized (lock) {
+            try {
+                if (!file.exists()) {
+                    file.createNewFile();
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Never use `printStackTrace` in real code, either rethrow the exception if the method permits it or log it to a `Logger`.